qiskit_experiments/calibration_management/base_calibration_experiment.py

```diff
@@ -0,0 +1,338 @@
+
+    This mixin class inherits from the :class:`BaseExperiment` class since calibration
+    experiments by default call :meth:`block_for_results`. This ensures that the next
+    calibration experiment cannot proceed before the calibration parameters have been
+    updated. Developers that wish to create a calibration experiment must subclass this
+    base class and the characterization experiment. Therefore, developers that use this
+    mixin class must pay special attention to their class definition. Indeed, the first
+    class should be this mixin and the second class should be the characterization
+    experiment since the run method from the mixin must be used. For example, the rough
+    frequency calibration experiment is defined as
+
+    .. code-block:: python
+
+        RoughFrequency(BaseCalibrationExperiment, QubitSpectroscopy)
+
+    This ensures that the :meth:`run` method of :class:`RoughFrequency` will be the
+    run method of the :class:`BaseCalibrationExperiment` class. Furthermore, developers
+    must explicitly call the :meth:`__init__` methods of both parent classes.
+
+    If the experiment uses custom schedules, which is typically the case, then
+    developers may chose to use the :meth:`get_schedules` method when creating the
+    circuits for the experiment. If :meth:`get_schedules` is used then the developer
+    must override at least one of the following methods used by :meth:`get_schedules`
+    to set the schedules:
+
+    #. :meth:`_get_schedules_from_options`
+
+    #. :meth:`_get_schedules_from_calibrations`
+
+    #. :meth:`_get_schedules_from_defaults`
+
+    These methods are called by :meth:`get_schedules`.
+
+    The :meth:`update_calibrations` method is responsible for updating the values of the parameters
+    stored in the instance of :class:`Calibrations`. Here, :class:`BaseCalibrationExperiment`
+    provides a default update methodology that subclasses can override if a more elaborate behaviour
+    is needed. At the minimum the developer must set the class variable :code:`__updater__` which
+    should have an :code:`update` method and can be chosen from the library
+    :mod:`qiskit_experiments.calibration_management.update_library`. See also
+    :class:`qiskit_experiments.calibration_management.update_library.BaseUpdater`. If no updater
+    is specified the experiment will still run but no update of the calibrations will be performed.
+
+    In addition to the calibration specific requirements, the developer must set the analysis method
+    with the class variable :code:`__analysis_class__` and any default experiment options.
+    """
+
+    # The updater class that updates the Calibrations instance. Different calibration
+    # experiments will use different updaters.
+    __updater__ = None
+
+    # pylint: disable=super-init-not-called
+    def __init__(
+        self,
+        calibrations: Union[BackendCalibrations, Calibrations],
```

Comment:
Is the `BackendCalibrations` necessary? It is subclass of `Calibrations`.

Reply:
Thanks, good catch. Fixed here: 5090893